### DISCUSSION TITLE
Window not centered

### DISCUSSION THREAD

**Discussion by Fortyseven** [beeware/toga#2196] [Debugging help]
Is there a way to force the window to be centered and placed on the primary monitor on start?

Currently, I'm not finding a rhyme or reason for where it shows up on my setup (vertically stacked multiple monitors, GNOME, Ubuntu 22.x).

Semi-related: https://github.com/beeware/toga/issues/1339

**Comment by freakboy3742** [comment]
You can position a window using the `window.position` API; however, we don't currently have an API to tell you what the size (and thus centre) of the screen would be. At present, the coordinate you provide is relative to the origin of the "primary" monitor; however, which monitor that is in a multi-screen setup, and how secondary screens relate to that origin is somewhat ill-defined.

#1930 is a proposed PR which would add a description of the screens accessible to an app. The broad API in that PR is likely to be accepted, but it still requires tests before it will be accepted into core.

As an aside, Linux in particular is problematic when it comes to describing the position of windows, because the window positioning is at the discretion of the window manager. The default window manager on Ubuntu and Fedora both allow for arbitrary window placement; but it's worth noting that this isn't a *requirement* for Linux window management. Window positions are, at best, hints to the window manager that may be ignored. If you're intending your app to have wide release, you shoudn't assume that window position will be honoured as part of your UI design.

**Reply by Fortyseven** [reply]
Fair enough; that certainly gives me a bigger picture of the situation. :weary:
